qiskit_experiments/base_experiment.py

```diff
@@ -288,3 +307,46 @@ def set_analysis_options(self, **fields):
             fields: The fields to update the options
         """
         self._analysis_options.update_options(**fields)
+
+    def _metadata(self) -> Dict[str, any]:
+        """Return experiment metadata for ExperimentData.
+
+        The :meth:`_add_job_metadata` method will be called for each
+        experiment execution to append job metadata, including current
+        option values, to the ``job_metadata`` list.
+        """
+        metadata = {
+            "experiment_type": self._type,
+            "num_qubits": self.num_qubits,
+            "physical_qubits": list(self.physical_qubits),
+            "job_metadata": [],
+        }
+        # Add additional metadata if subclasses specify it
+        for key, val in self._additional_metadata():
+            metadata[key] = val
+        return metadata
+
+    def _additional_metadata(self) -> Dict[str, any]:
+        """Add additional subclass experiment metadata.
+
+        Subclasses can override this method if it is necessary to store
+        additional experiment metadata in ExperimentData.
+        """
+        return {}
+
+    def _add_job_metadata(self, experiment_data: ExperimentData, job: BaseJob, **run_options):
+        """Add runtime job metadata to ExperimentData.
+
+        Args:
+            experiment_data: the experiment data container.
+            job: the job object.
+            run_options: backend run options for the job.
+        """
+        metadata = {
+            "job_id": job.job_id(),
+            "experiment_options": copy.copy(self.experiment_options.__dict__),
+            "transpile_options": copy.copy(self.transpile_options.__dict__),
+            "analysis_options": copy.copy(self.analysis_options.__dict__),
+            "run_options": copy.copy(run_options),
+        }
+        experiment_data._metadata["job_metadata"].append(metadata)
```

Comment:
Out of curiosity: why not use the `job_id` as key?

Reply:
An older version had that, but it was difficult to use since you didn't necessarily know the job id without looking up other methods, and didn't easily know what order jobs were run. With a list you can do `["job_metadata"][-1]` to get the most recently run data.

Reply:
Nice.